src/extension.rs

```diff
@@ -47,6 +47,17 @@ impl ExtensionRegistry {
 /// An Extension Registry containing no extensions.
 pub const EMPTY_REG: ExtensionRegistry = ExtensionRegistry::new();
 
+impl<T: IntoIterator<Item = Extension>> From<T> for ExtensionRegistry {
+    fn from(value: T) -> Self {
+        let mut reg = Self::new();
+        for ext in value.into_iter() {
+            let prev = reg.0.insert(ext.name.clone(), ext);
+            assert!(prev.is_none());
```

Comment:
Could be a try from, or at least add a message to the assert

Reply:
Done the latter, try-from leads to a conflicting instance error, or else one about uncovered type params. (There's blanket TryFrom = `From` + `Infallible` thrown into the mix.)